cmd/botkube/main.go

```diff
@@ -56,97 +57,156 @@ const (
 )
 
 func main() {
+	if err := run(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+// run wraps the main logic of the app to be able to properly clean up resources via deferred calls.
+func run() error {
```

Comment:
That's a way to make sure the deferred call with closing analytics reporter is properly executed before exiting the app. I find it the best looking option. Other alternative is e.g. doing deferred call with `os.Exit` taking an exit code which is set instead of using `log.Fatal`.

Reply:
I'm ok with it 👍